src/pal/src/exception/signal.cpp

```diff
@@ -67,56 +70,67 @@ typedef void *siginfo_t;
 #endif  /* !HAVE_SIGINFO_T */
 typedef void (*SIGFUNC)(int, siginfo_t *, void *);
 
+#if !HAVE_MACH_EXCEPTIONS
 // Return context and status for the signal_handler_worker.
 struct SignalHandlerWorkerReturnPoint
 {
     bool returnFromHandler;
     CONTEXT context;
 };
+#endif // !HAVE_MACH_EXCEPTIONS
 
 /* internal function declarations *********************************************/
 
+#if !HAVE_MACH_EXCEPTIONS
 static void sigill_handler(int code, siginfo_t *siginfo, void *context);
 static void sigfpe_handler(int code, siginfo_t *siginfo, void *context);
 static void sigsegv_handler(int code, siginfo_t *siginfo, void *context);
 static void sigtrap_handler(int code, siginfo_t *siginfo, void *context);
 static void sigbus_handler(int code, siginfo_t *siginfo, void *context);
 static void sigint_handler(int code, siginfo_t *siginfo, void *context);
 static void sigquit_handler(int code, siginfo_t *siginfo, void *context);
+#endif // !HAVE_MACH_EXCEPTIONS
 static void sigterm_handler(int code, siginfo_t *siginfo, void *context);
```

Comment:
A nit - could you please move this line above all of the handler declarations? Then you can use single `#if !HAVE_MACH_EXCEPTIONS`